content/departments/cloud/technical-docs/v1.1/mi1-1_creation_process.md

```diff
@@ -4,83 +4,53 @@ Creating a new [managed instance](./index.md) involves following the steps below
-   - `export TF_VAR_opsgenie_webhook=<OpsGenie Webhook value>`
-     - This can be found in the [Managed Instances vault](https://my.1password.com/vaults/nwbckdjmg4p7y4ntestrtopkuu/allitems/d64bhllfw4wyybqnd4c3wvca2m)
+- `customer` - name of customer
+- `ce_email` - email of Customer Engineer from issue
+- `customer_email` - customer admin email (only one from provided in issue)
+- `instance_type` - purpose of this instance
 
-1. Check out a new branch: `git checkout -b $COMPANY/create-instance`
-1. Ensure the target version of docker-compose file is in the golden directory, it should be named `docker-compose.x.y.z.yaml`
-1. ` ./util/create-managed-instance-new.sh $COMPANY` and **commit the result**. Make sure that the version exists in [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker/tags).
-1. Replace base `docker-compose.yaml` to use golden symlink: `cd $COMPANY/red/docker-compose && rm docker-compose.yaml && ln -s ../../../golden/docker-compose.<MAJOR.MINOR.PATCH>.yaml docker-compose.yaml && rm ../VERSION && cd ../../`
-1. Open and edit `terraform.tfvars` according to the TODO comments within and commit the result.
+      trial - for customer trial
 
-   > NOTE: ⚠️ Do not set `enable_alerting` to `true` yet as this will cause false alerts to fire until the MI creation process has been completed!
+      production - for paying customer
 
-1. Open and edit `deploy-sourcegraph-managed/$COMPANY/red/docker-compose/docker-compose.override.yaml`, increase `gitserver-0`'s `cpus: 8` if the instance size is larger than "n2-standard-8".
-1. In `deploy-sourcegraph-managed/$COMPANY/project` run `terraform init && terraform apply -var-file=../terraform.tfvars` to create the GCP project
-1. In `deploy-sourcegraph-managed/$COMPANY` run `terraform init && terraform apply`
-1. Check if all is running
+      internal - for internal Sourcegraph usage
 
-   ```bash
-   mg check
-   ```
+  via command line:
 
-1. In the `$COMPANY` directory run `terraform fmt` and commit any changes
+  ```
+  gh workflow run mi_create.yml -f customer=$CUSTOMER -f ce_email=$CE_EMAIL -f customer_email=$CUSTOMER_EMAIL instance_type=[production|trial|internal]
+  ```
 
-1. In the `$COMPANY` GCP project, create [Google Oauth credentials](https://console.cloud.google.com/apis/credentials?project=sourcegraph-managed-$COMPANY) with the following parameters:
+  or via [GithubAction web console](https://github.com/sourcegraph/deploy-sourcegraph-managed/actions/workflows/mi_create.yml)
 
-   - type: Web Application
-   - name: `managed-instance-$COMPANY`
-   - Authorized redirect URIs:
-     - Is the instance **public**, then add **only** `https://$PROJECT_SLUG/.auth/callback`
-     - Is the instance **private**, then add **both** `https://$PROJECT_SLUG/.auth/callback` AND `http://localhost/.auth/callback`
+  Important: GithubAction is `idempotent`, so can be safely re-run multiple times.
```

Comment:
Is it idempotent over the full input, or only certain fields (ex. if I make a mistake in CE_EMAIL and fix it, will it create a new instance or will it edit the current one)?

Reply:
You can safely rerun with same parameters, if you change fields (apart from customer name), it will either add (ce email), nor do thing for different customer email nor label chnaged.